src/dataTypes/DashboardItems/DashboardItemSmallType.tsx

```diff
@@ -0,0 +1,9 @@
+import { ReactNode } from 'react';
+
+export type DashboardItemSmallType = {
+    title: string;
+    titleSize: 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900;
+    width: number | string;
+    height: number | string;
+    children: ReactNode;
+};
```

Comment:
A few things worth mentioning here, so this will apply to more than just the lines mentioned;

- Files that do not contain Jsx content (e.g. `<Component>`) should have a `.ts` ending to clearly separate the two
- The folder name `dataTypes` could be simplified to `types`, or update to match the `mockdata` directory (all lowercase vs camelCase`
- Do you need a folder for the `DashboardItems`? There are only two very short files under it, so they could be merged into one file under (e.g.) `types/DashboardItem.ts`
- Types should not be suffixed by `Type` as this is implicit
- There is quite some overlap between the two types, and `DashboardItemSmallType` is a strict subset of `...LargeType`, so we can use some TS magic here to make the code DRYer (DRY=Do not Repeat Yourself):
```ts
export type DashboardItemSmall = {
  title: string;
  titleSize: 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900;
  width: number | string;
  height: number | string;
}

export type DashboardItemLarge = DashboardItemSmall & {
  paragraph: string;
  textSize: 300 | 400 | 500;
}
```
Here you might find `interfaces` more intuitive, as they are written as;
```ts
interface IDashboardItemSmall {
  // ...
}

interface IDashboardItemLarge extends IDashboardItemSmall {
 // extra properties...
}
```
- Don't include children in your props. The definition of children here also doesn't encompass all acceptable React.children values. I don't know if this is intentional or not, but it might be worth noting. When you use the format of `React.FC<Props>`, `children` is automatically included from the generic `FC` (Functional Component) type.
- Generally speaking, if you wish to pass _one_ item of a type as a prop, it's better to use _a_ prop with that type instead of exploding the entire item into the props. Better demonstrated with an example;
```tsx
const Test: React.FC<{ name: string, age: number, pet: Animal }> = ({ name, age, pet }) => {
  // ...
}

// versus the more compact

const Test: React.FC<{ person: Person }> = ({ person }) => {
  // refer to person.name, person.age, &c
}
```
The benefits of this show when passing in the props;

```tsx
<Test name={person.name} age={person.age} pet={person.pet} />

// vs

<Test person={person} />
```

Consider [the line in this PR](https://github.com/IT2901-SINTEF01/frontend/pull/45/files#diff-d76616b8eab54f7d30f20cd8ccd132b403adf3a97e645f5c7b3c0e05d29b8c9aR9);
```tsx
const Template: Story<DashboardItemLargeType> = (args) => <DashboardItemLarge {...args} />;
```
What happens if there isn't a one-to-one relationship between the generic type `Story<DashoardItemLargeType>` and the expected props for `DashboardItemLarge`?

I would argue it's more canonical with;

```tsx
const Template: Story<ItemLarge> = (args) => (
  <DashboardItemLarge item={args.item} /> // or create item if necessary
);
```

Reply:
Regarding the last lines;

You might need to look into `PropsWithChildren` (or something along those lines) for the generic `Story` type.

Reply:
I will look into this!